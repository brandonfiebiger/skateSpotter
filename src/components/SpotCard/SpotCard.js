import React, { Component } from 'react';
import {
  View,
  Button,
  Text,
  Image,
  StyleSheet,
  TouchableHighlight
} from 'react-native';

export default class SpotCard extends Component {
  render() {
    const { name, description, image } = this.props;

    return (
      <View style={styles.card}>
        <View style={styles.imageContainer}>
          <Image style={styles.image} source={image} />
        </View>
        <View style={styles.contentContainer}>
          <Text style={[styles.header, styles.text]}>{name}</Text>
          <Text style={[styles.description, styles.text]}>{description}</Text>
          <Text style={[styles.distance, styles.text]}>
            Distance From Me: 4 mi.
          </Text>
          <TouchableHighlight style={styles.routeButton}>
            <Button color="#f7f7f7" title="Route" />
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  card: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    height: 700,
    width: '100%'
  },
  imageContainer: {
    height: '50%',
    width: '100%'
  },
  contentContainer: {
    flex: 1,
    flexDirection: 'column',
    fontSize: 5,
    // justifyContent: 'space-around',
    height: '100%',
    width: '100%'
  },
  image: {
    height: '100%',
    width: '100%'
  },
  routeButton: {
    backgroundColor: 'black',
    borderRadius: 10,
    height: 35,
    opacity: 0.8,
    paddingBottom: 5,
    width: 80
  },
  header: {},
  description: {},
  distance: {},
  text: {}
});
