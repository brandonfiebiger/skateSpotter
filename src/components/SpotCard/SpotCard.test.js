import { SpotCard } from './SpotCard';
import 'react-native';
import React from 'react';


import renderer from 'react-test-renderer';

let tree;
beforeEach(() => {
  tree = renderer.create(
    <SpotCard /> 
  ).toJSON();
})

describe('SpotCard', () => {

  it('should match the snapshot', () => {
    expect(tree).toMatchSnapshot()
  })
})