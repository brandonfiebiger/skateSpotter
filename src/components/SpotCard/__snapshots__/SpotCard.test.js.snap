// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotCard should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": "50%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "height": "50%",
        "width": "50%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/skate-spotter-home.png",
        }
      }
      style={
        Object {
          "height": "50%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "height": "50%",
        "width": "50%",
      }
    }
  >
    <Text>
      Spot Name
    </Text>
    <Text>
      Description
    </Text>
    <Text>
      Distance From Me:
    </Text>
  </View>
</View>
`;
